lib/board.rb

```diff
@@ -0,0 +1,62 @@
+class Board
+
+  attr_accessor :cells
+
+  def initialize
+    self.cells = [" ", " ", " ", " ", " ", " ", " ", " ", " "]
+  end
+
+  def reset!
+    @cells = Array.new(9, " ")
+    @cells
+  end
+
+def display
+    puts " "
+    puts "     Here is the current board:"
+    puts " "
+    puts "      #{self.cells[0]} | #{self.cells[1]} | #{self.cells[2]} "
+    puts "     -----------"
+    puts "      #{self.cells[3]} | #{self.cells[4]} | #{self.cells[5]} "
+    puts "     -----------"
+    puts "      #{self.cells[6]} | #{self.cells[7]} | #{self.cells[8]} "
+  end
+
+
+  def position(number)
+    self.cells[number.to_i - 1]
+  end
+
+  def full?
+    return true if self.cells.none? { |position| position == " " }
+  end
+
+  def turn_count
+  self.cells.select {|position| position != " "}.count
```

Comment:
Use the highest level and most expressive iterator possible to solve a problem. [`count`](http://ruby-doc.org/core-2.2.0/Array.html#method-i-count) on the cells is perfect and the rest of this method is not needed. Entire body could read:

```ruby
def turn_count
  self.cells.count {|position| position != " "}
end
```